### PR TITLE
chore(trie): replace ParallelStateRoot with AsyncStateRoot

### PR DESCRIPTION
**Pull request by fgimenez**
In #10927  `tokio::sync::oneshot` channels used for synchronization in `AsyncStateRoot` were replaced by sync versions because we were awaiting results in order and the overhead of setting up and using the tokio runtime was not giving any benefit. With these changes, `AsyncStateRoot` is no longer async.

Given that `ParallelStateRoot` is not used and that `AsyncStateRoot` is an optimization over it keeping its parallel nature, in this PR the current `ParallelStateRoot` is removed, and  `AsyncStateRoot` is renamed to `ParallelStateRoot`.

### CODE REVIEW

**Review by mattsse — APPROVED** [review]
nice

File: crates/engine/tree/src/tree/mod.rs [hunk 1]
```diff
@@ -2264,20 +2264,18 @@ where
...
     ///
     /// # Returns
     ///
     /// Returns `Ok(_)` if computed successfully.
     /// Returns `Err(_)` if error was encountered during computation.
     /// `Err(ProviderError::ConsistentView(_))` can be safely ignored and fallback computation
-
-    /// should be used instead.
-    fn compute_state_root_async(
+    fn compute_state_root_parallel(
```
Comment (DaniPopes) [inline crates/engine/tree/src/tree/mod.rs:2274]: last doc line got dropped

Comment (fgimenez) [reply crates/engine/tree/src/tree/mod.rs:2275]: good catch, thx!

**Review by DaniPopes — APPROVED** [review]

File: crates/trie/parallel/Cargo.toml [hunk 2]
```diff
@@ -58,12 +55,11 @@ proptest.workspace = true
 proptest-arbitrary-interop.workspace = true
 
 [features]
-default = ["metrics", "async", "parallel"]
+default = ["metrics", "parallel"]
 metrics = ["reth-metrics", "dep:metrics", "reth-trie/metrics"]
-async = ["tokio/sync", "itertools"]
-parallel = ["rayon"]
+parallel = ["rayon", "itertools"]
```
Comment (rkrasiuk) [inline crates/trie/parallel/Cargo.toml:60]: let's get rid of this feature, since this is the only impl that crate exposes

**Review by rkrasiuk — APPROVED** [review]
lgtm pending feature removal

File: crates/trie/parallel/Cargo.toml (hunk 2 above)
Comment (fgimenez) [reply crates/trie/parallel/Cargo.toml]: sure done here https://github.com/paradigmxyz/reth/pull/11213/commits/456db3b088220e728c2e801552f13d58797341ca

**Pull request merged**